### ISSUE TITLE
update certificates

### ISSUE THREAD

**Issue by ragnar76** [freemint/git#3]
Hi there, the certificates are expired. Do you plan to update them or how can i get new one?

**Comment by mikrosk** [comment]
You are right. Not only them but the whole package should be updated. To this day I haven't heard of anyone actually using `git` on Atari, so that's the reason why it is in such shape.

**Comment by ragnar76** [comment]
Well, together with Latz we want to try out whether it is worth developing the Firebee setup further via git (https://codeberg.org/firebee/FireBee_Setup). For this we would need git on the Atari or the Firebee. It doesn't make sense to copy the setup back and forth over the network and then move it to a repository via git.

I'm not that deep into git, does it accept the Mozilla cert collection, or the summary of all of them (like cacert.pem)?

**Comment by mikrosk** [comment]
That's a nice idea. However be aware of few things:
- git works with long names, it stores its commits as SHA hash-based filenames 
- it destroys original timestamps
- for simple download you don't really need git. not sure about your codeberg but in github you can easily access its repository archives, e.g. https://github.com/freemint/freemint/archive/refs/heads/master.zip (this takes the master branch of the freemint project, zip it and download it)

If I were you, I'd maintain a git tree on my Linux machine and used FireBee only for downloading the finished product. I presume you don't _create_ so much content on FireBee which would need transferring back to the host machine + git commit.

> does it accept the Mozilla cert collection

I think this is where I took it from, yes.

**Comment by ragnar76** [comment]
> git works with long names, it stores its commits as SHA hash-based filenames

does it really matters? i mean, most of freemints files are 8+3 too

> it destroys original timestamps

I don't care :D . Does it matter too? While moving files back and forth, the files always get a new timestampg.

>for simple download ...

Yes, the "regular" user can download a release (or the mentioned master.zip) as a http download (or a [scripted] copy to firebee.org as ftp).

> If I were you, I'd maintain a git tree on my Linux machine and used FireBee only for downloading the finished product. I presume you don't create so much content on FireBee which would need transferring back to the host machine + git commit.

In my mind, only the devs are using git. For example, for the included source files like freemint or emutos, we can make use of submodules, use different branches (for example, we can test freemints new video settings in xaaes without changing the prod branch) and all the others bells and whistles.

> I think this is where I took it from, yes.

Ok, will try to use them this evening

**Comment by mikrosk** [comment]
> does it really matters?

It does, if you don't want to start losing commits. 1234567890 and 1234567891 would become identical files.

> In my mind, only the devs are using git

And devs are usually using the host machine. All of the mentioned you can do on the host machine and then just overwrite the filesystem on the FireBee, without caring about stuff in `.git` folder.

But that's only my suggestion of course, do as you feel appropriate for your needs.

**Comment by ragnar76** [comment]
Okay, got it to work after updating the ca-certificates. It was quite easy:
- on the firebee, remove content from /usr/m68k-atari-mint/certs/*
- copy content from /usr/share/ca-certificates/mozilla/* to the same location on the firebee
- copy /etc/ssl/certs/* to /usr/m68k-atari-mint/certs on the firebee

i'd like to make package for this repo, whats the best way to do it? I can't create a pull request on a tagged file, can i?

![a screenshot of a desktop. git is running in a terminal](https://files.mastodon.social/media_attachments/files/109/253/848/460/692/364/original/ec1b035cbd268ba3.jpeg "a screenshot of a desktop. git is running in a terminal")

**Comment by mikrosk** [comment]
Oh nice! For now, just attach them here, if/when I find some time, I'll try to make a new git build as well. I'll leave this issue open for now.

**Comment by ragnar76** [comment]
Here we go
[ca-certs.tar.gz](https://github.com/freemint/git/files/9894734/ca-certs.tar.gz)

Isn't it possible to implement a function into travis (if you still use it) to download and make a packe or convert, for example, the debian package here (https://packages.debian.org/bullseye/ca-certificates)

**Comment by mikrosk** [comment]
Travis is no longer an option but sure, Github Action allows it. It's just matter of time and motivation. :-/
